### PR TITLE
fix: drop IDB chunk store

### PR DESCRIPTION
**Pull request by ThaUnknown**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[ ] Bug fix
[ ] New feature
[x] Other, please explain:

**What changes did you make? (Give an overview)**
Dropped hybrid chunk store, to use FSA chunk store directly, this drops IDB support.

Lets assume every1 updates their browsers, as all browsers since recently support OPFS https://caniuse.com/mdn-api_storagemanager_getdirectory putting us at 88% coverage.

actually safari doesn't support createWritable https://caniuse.com/mdn-api_filesystemfilehandle_createwritable so this is a 73% coverage :/ maybe lets leave this PR to sit for a while

FSA offers waaaaaay more storage capacity, way better speeds and way smaller bundle sizes.

### CODE REVIEW

**Comment by ThaUnknown** [comment]
I guess we can merge this once https://bugs.webkit.org/show_bug.cgi?id=254726 is resolved, since safari is probably the biggest block on this

File: lib/torrent.js
Comment (ThaUnknown) [inline lib/torrent.js:65]: this is a hellscape, how to do this better?

Comment (SilentBot1) [reply lib/torrent.js:65]: This could become:
```js
const FALLBACK_STORE = process.browser &&
  navigator?.storage?.getDirectory &&
  FileSystemFileHandle?.prototype?.createWritable
    ? FSChunkStore
    : MemoryChunkStore;
 ```
But it does assume that in the browser that `navigator` and `FileSystemFileHandle` are both defined, failing that, the undefined checks will still need to exist.

Comment (ThaUnknown) [reply lib/torrent.js:65]: this has potential to fail in node...
i considered
```js
const FALLBACK_STORE = !process.browser
  ? FSChunkStore
  : (typeof navigator !== 'undefined' && navigator.storage?.getDirectory && typeof FileSystemFileHandle !== 'undefined' && FileSystemFileHandle?.prototype?.createWritable)
    ? FSChunkStore // FSAChunkStore
    : MemoryChunkStore

Comment (ThaUnknown) [reply lib/torrent.js:65]: or a separate 
```js
const SUPPORTS_FSA = (typeof navigator !== 'undefined' && navigator.storage?.getDirectory && typeof FileSystemFileHandle !== 'undefined' && FileSystemFileHandle?.prototype?.createWritable)

const FALLBACK_STORE = !process.browser
  ? FSChunkStore // FSChunkStore - nodejs/bun
  : SUPPORTS_FSA
    ? FSChunkStore // FSAChunkStore - browser
    : MemoryChunkStore
```

Comment (SilentBot1) [reply lib/torrent.js:65]: imo separate would be better at that point

**Review by SilentBot1 — APPROVED** [review]

**Pull request merged**
